src/huggingface_hub/hub_mixin.py

```diff
@@ -37,34 +39,59 @@ def save_pretrained(
         """
         Save weights in local directory.
 
-                Parameters:
-                    save_directory (`str`):
-                        Specify directory in which you want to save weights.
-                    config (`dict`, *optional*):
-                        specify config (must be dict) in case you want to save
-                        it.
-                    push_to_hub (`bool`, *optional*, defaults to `False`):
-                        Set it to `True` in case you want to push your weights
-                        to huggingface_hub
-                    kwargs (`Dict`, *optional*):
-                        kwargs will be passed to `push_to_hub`
+        Parameters:
+            save_directory (`str`):
+                Specify directory in which you want to save weights.
+            config (`dict`, *optional*):
+                specify config (must be dict) in case you want to save
+                it.
+            push_to_hub (`bool`, *optional*, defaults to `False`):
+                Whether or not to push your model to the Hugging Face model hub after
+                saving it. You can specify the repository you want to push to with
+                `repo_id` (will default to the name of `save_directory` in your
+                namespace).
+            kwargs:
+                Additional key word arguments passed along to the
+                [`~utils.PushToHubMixin.push_to_hub`] method.
         """
-
         os.makedirs(save_directory, exist_ok=True)
 
+        # saving model weights/files
+        self._save_pretrained(save_directory)
+
         # saving config
         if isinstance(config, dict):
             path = os.path.join(save_directory, CONFIG_NAME)
             with open(path, "w") as f:
                 json.dump(config, f)
 
-        # saving model weights/files
-        self._save_pretrained(save_directory)
-
         if push_to_hub:
-            return self.push_to_hub(save_directory, **kwargs)
-
-    def _save_pretrained(self, save_directory):
+            kwargs = kwargs.copy()  # soft-copy to avoid mutating input
+            if config is not None:  # kwarg for `push_to_hub`
+                kwargs["config"] = config
+
+            if (
+                # If a deprecated argument is passed, we have to use the deprecated
+                # version of `push_to_hub`.
+                # TODO: remove this possibility in v0.12
+                kwargs.get("repo_url") is not None
+                or kwargs.get("repo_path_or_name") is not None
+                or kwargs.get("organization") is not None
+                or kwargs.get("use_auth_token") is not None
+                or kwargs.get("git_user") is not None
+                or kwargs.get("git_email") is not None
+                or kwargs.get("skip_lfs_files") is not None
+            ):
+                if kwargs.get("repo_path_or_name") is None:
```

Comment:
Should we have a deprecation warning here or do we think with `push_to_hub` warning it's enough?

Reply:
I can deprecate the argument in `save_pretrained` as well but I thought that since `save_pretrained` is just forwarding the kwargs to `push_to_hub` it was fine to warn only once.

But I get your point so if we really want to deprecate in `save_pretrained` as well, I really don't mind adding the decorator here as well.

Reply:
No strong opinion, up to you!

Reply:
Ok, I'll keep it as it is then.